src/Server/GRPCServer.cpp

```diff
@@ -484,19 +477,12 @@ namespace
                   grpc::ServerCompletionQueue & notification_queue,
                   const CompletionCallback & callback) override
         {
-            grpc_service.RequestExecuteQueryWithStreamOutput(&grpc_context, &query_info.emplace(), &writer, &new_call_queue, &notification_queue, getCallbackPtr(callback));
+            grpc_service.RequestExecuteQueryWithStreamOutput(&grpc_context, &query_info, &writer, &new_call_queue, &notification_queue, getCallbackPtr(callback));
         }
 
         void read(GRPCQueryInfo & query_info_, const CompletionCallback & callback) override
         {
-            if (!query_info.has_value())
-            {
-                callback(false);
-                return;
-            }
-
-            query_info_ = std::move(query_info).value();
-            query_info.reset();
+            query_info_ = std::move(query_info);
```

Comment:
After the `std::move` the state of `query_info` depends on the implementation of the move assignment operator of the third-party`grpc::QueryInfo` type.

Btw, C++ standard guaranties a valid but unspecified state of the moved-from object. This is why I believe std::optional was used in the first place.